### PR TITLE
rewrite of labrecorder in C++

### PR DESCRIPTION
**Pull request by dmedine**
Rewrite of LabRecorder in C++. Thus far, it looks and feels exactly like the Python version did. The one exception is scripted action functionality has not been implemented (since it is no longer in python), but there are plans to re-implement this as a lua interpreter. Also coming soon is config options for automatic stream syncing.

I believe this will record correctly as that part of the code is the same. The only changes were on the interface level. Preliminary tests show that this is true.

### CODE REVIEW

**Comment by dmedine** [comment]
So far so good.The recording code itself is identical to the old version of labrecorder, so there should be no problem there. As far as I can tell, the gui part (now in C++) works as it should.

**Comment by dmedine** [comment]
I also found a typo in the AMTI force plate code. This also go merged.

File: Apps/AMTIForcePlate/mainwindow.cpp [hunk 1]
```diff
@@ -281,7 +281,7 @@ void MainWindow::read_thread(HANDLE hPort, int comPort, int samplingRate, int nu
 			.append_child_value("type","MomentumX");
 		channels.append_child("channel")
 			.append_child_value("label",("My" + deviceTag[k]).c_str())
-			.append_child_value("type","MomentumZ");
+			.append_child_value("type","Momentumy");
```
Comment (chkothe) [inline Apps/AMTIForcePlate/mainwindow.cpp]: May want to capitalize that for consistency.

Comment (dmedine) [inline Apps/AMTIForcePlate/mainwindow.cpp]: Akk! Good eye. Will do.

**Pull request merged**